packages/react-ui/components/Select/Select.tsx

```diff
@@ -664,15 +667,15 @@ export class Select<TValue = {}, TItem = {}> extends React.Component<SelectProps
   };
 }
 
-function renderValue(value: any, item: any) {
+function renderValue<TValue, TItem>(value: TValue, item: Nullable<TItem>) {
   return item;
 }
 
-function renderItem(value: any, item: any) {
+function renderItem<TValue, TItem>(value: TValue, item?: TItem | undefined) {
```

Comment:
```suggestion
function renderItem<TValue, TItem>(value: TValue, item?: TItem) {
```

В тип опционального элемента автоматически добавляется `undefined`.